### PR TITLE
:bug: Exclude non-required questionnaires in assessed calculation

### PR DESCRIPTION
**Pull request by ibolton336**
Resolves https://issues.redhat.com/browse/MTA-1967

Addresses an issue where the assessed boolean for archetypes is set to true when all questionnaires are marked as not-required. Uses the new assessment.required boolean surfaced in https://github.com/konveyor/tackle2-hub/commit/25522337608b2dbab5df85ce4a5b5a9fc78a2289 to filter out these false positives.

### CODE REVIEW

File: client/src/app/queries/assessments.ts [hunk 1]
```diff
@@ -262,3 +264,33 @@ export const useFetchAssessmentsWithArchetypeApplications = () => {
...
   };
 };
+
+export const useFetchAllAssessmentsWithArchetypes = (
+  archetypes: Archetype[]
+) => {
+  const assessmentQueries = useQueries({
+    queries: archetypes.map((archetype) => ({
+      queryKey: ["assessmentsForArchetype", archetype.id],
+      queryFn: () => getAssessmentsByItemId(true, archetype.id), // Replace with actual API call
```
Comment (sjd78) [inline client/src/app/queries/assessments.ts:274]: Is this comment still valid?

Comment (sjd78) [reply client/src/app/queries/assessments.ts:274]: Also, kinda a bummer we can't just send a single request to "get assessments for this list of archetype ids".

File: client/src/app/pages/applications/components/application-detail-drawer/components/assessed-archetypes.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+export const AssessedArchetypes: React.FC<IAssessedArchetypesProps> = ({
+  archetypeRefs,
+}) => {
+  const { t } = useTranslation();
+  const { archetypes, isFetching: isFetchingArchetypes } = useFetchArchetypes();
+  const applicationArchetypes = archetypes.filter(
+    (archetype) => archetypeRefs?.some((ref) => ref.id === archetype.id)
+  );
```
Comment (sjd78) [inline client/src/app/pages/applications/components/application-detail-drawer/components/assessed-archetypes.tsx:20]: Just a thought -- this kind of thing could be embedded directly in the hook so the hook use could look like:
```js
  const { archetypes, ... } = useFetchArchetypes({ 
    includeIds: archetypeRefs.map((a) => a.id) 
  });
```

or even somethings more like:
```js
  const { archetypes, ... } = useFetchArchetypes({
    forApplication: application
  });
```

Then if every the rest endpoint gets a filter itself, that can rework can be handled directly in the query hook.

File: client/src/app/pages/applications/components/application-detail-drawer/components/assessed-archetypes.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  const {
+    assessmentsWithArchetypes,
+    isLoading: isFetchingAllAssessmentsWithArchetypesLoading,
+  } = useFetchAllAssessmentsWithArchetypes(applicationArchetypes || []);
```
Comment (sjd78) [inline client/src/app/pages/applications/components/application-detail-drawer/components/assessed-archetypes.tsx:25]: The `|| []` is easier to do as a default value in the hook function.

File: client/src/app/queries/assessments.ts [hunk 4]
```diff
@@ -262,3 +264,33 @@ export const useFetchAssessmentsWithArchetypeApplications = () => {
     isLoading: assessmentsLoading || isArchetypesLoading,
   };
 };
+
+export const useFetchAllAssessmentsWithArchetypes = (
+  archetypes: Archetype[]
```
Comment (sjd78) [inline client/src/app/queries/assessments.ts:269]: Can just have `[]` as the default value and allow the param to be optional:
```suggestion
  archetypes: Archetype[] = []
```

File: client/src/app/pages/applications/components/application-detail-drawer/components/assessed-archetypes.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  const filteredArchetypes = assessmentsWithArchetypes
```
Comment (sjd78) [inline client/src/app/pages/applications/components/application-detail-drawer/components/assessed-archetypes.tsx:26]: These are assessed archetypes where at least one of the assessments is required.  I think this matches the PR description, but it was a bit confusing at first look.

**Review by sjd78 — COMMENTED** [review]
A few comments and thoughts

File: client/src/app/pages/applications/components/application-detail-drawer/components/assessed-archetypes.tsx (hunk 2 above)
Comment (ibolton336) [reply client/src/app/pages/applications/components/application-detail-drawer/components/assessed-archetypes.tsx:20]: Interesting - I wonder if the api supports this?

**Pull request merged**